dll/win32/shell32/wine/shellpath.c

```diff
@@ -713,41 +754,37 @@ BOOL WINAPI PathResolveW(LPWSTR path, LPCWSTR *dirs, DWORD flags)
             if (!PathIsAbsoluteW(path))
                 return PathMakeAbsoluteW(path) && PathFileExistsAndAttributesW(path, NULL);
 #else
-            return TRUE;
+            return TRUE; /* Found */
 #endif
         }
+
+        return FALSE; /* Not found */
     }
-    else if (!PathIsURLW(path))
-    {
-        if (flags & PRF_FIRSTDIRDEF)
-            PathQualifyExW(path, *dirs, 1);
-        else
-            PathQualifyExW(path, NULL, 1);
 
-        if (flags & PRF_VERIFYEXISTS)
+    if (PathIsURLW(path)) /* URL? */
+        return FALSE;
+
+    /* Qualify the path */
+    PathQualifyExW(path, ((flags & PRF_FIRSTDIRDEF) ? *dirs : NULL), 1);
+
+    if (flags & PRF_VERIFYEXISTS) /* Verify the existence? */
+    {
+        if ((flags & PRF_TRYPROGRAMEXTENSIONS) && /* Try to find a program? */
+            PathSearchOnExtensionsW(path, dirs, FALSE, dwWhich)) /* Search it */
         {
-            if ((flags & PRF_TRYPROGRAMEXTENSIONS) &&
-                PathSearchOnExtensionsW(path, dirs, FALSE, dwWhich))
-            {
-                return TRUE;
-            }
-            else if (!PathFileExistsAndAttributesW(path, NULL))
-            {
-                return FALSE;
-            }
+            return TRUE; /* Found */
         }
 
+        if (!PathFileExistsAndAttributesW(path, NULL)) /* Check the existence */
+            return FALSE; /* Not found */
+    }
+
 #if (_WIN32_WINNT >= _WIN32_WINNT_VISTA)
```

Comment:
This exists also on Win2k3 SP2.
Also all this block (surrounded by the #if ...) seems to be done only if `flags & PRF_VERIFYEXISTS` is TRUE.

Reply:
Used `_WIN32_WINNT_WS03`.
Moved code into `if (flags & PRF_VERIFYEXISTS)` block.